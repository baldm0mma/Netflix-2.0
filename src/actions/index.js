export const updateMovies = (movies, genre) => ({
  type: 'UPDATE_MOVIES',
  payload: {
    movies,
    genre
  }
});

export const chooseGenre = genre => ({
  type: 'CHOOSE_GENRE',
  payload: {
    genre
  }
});

export const chooseMovie = (id) => ({
  type: 'CHOOSE_MOVIE',
  payload: {
    id
  }
});

export const toggleFavorite = (id, favorited) => ({
  type: 'TOGGLE_FAVORITE',
  payload: {
    id,
    favorited
  }
});

export const setFavorites = favorites => ({
  type: 'SET_FAVORITES',
  payload: {
    favorites
  }
});

export const signIn = user => ({
  type: 'SIGN_IN',
  payload: {
    user
  }
});

export const signOut = () => ({
  type: 'SIGN_OUT'
});

export const throwError = error => ({
  type: 'THROW_ERROR',
  payload: {
    error
  }
});

export const isLoading = isLoading => ({
  type: 'IS_LOADING',
  payload: {
    isLoading
  }
});

export const searchQuery = searchTerm => ({
  type: 'SEARCH_QUERY',
  payload: {
    searchTerm
  }
});

export const addMovie = movie => ({
  type: 'ADD_FAVORITE',
  payload: {
    movie
  }
});