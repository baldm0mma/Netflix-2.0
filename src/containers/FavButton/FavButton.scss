// .favorite__img-button:hover {
//   background-image: url('../../images/like-filled.png');
// }

// .favorite__img-button{
//   background-image: url('../../images/like-empty.png');
//   z-index:400000000000;
// }