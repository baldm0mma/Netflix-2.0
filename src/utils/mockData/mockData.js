export const mockMovies = {
  popular: {
    "id": 420818,
    "video": false,
    "vote_average": 7.1,
    "title": "The Lion King",
    "popularity": 636.278,
    "poster_path": "\/dzBtMocZuJbjLOXvrl4zGYigDzh.jpg",
    "original_language": "en",
    "original_title": "The Lion King",
    "genre_ids": [
      12,
      16,
      10751,
      18,
      28
    ],
    "backdrop_path": "\/1TUg5pO1VZ4B0Q1amk3OlXvlpXV.jpg",
    "adult": false,
    "overview": "Simba idolises his father, King Mufasa, and takes to heart his own royal destiny. But not everyone in the kingdom celebrates the new cub's arrival. Scar, Mufasa's brother—and former heir to the throne—has plans of his own. The battle for Pride Rock is ravaged with betrayal, tragedy and drama, ultimately resulting in Simba's exile. With help from a curious pair of newfound friends, Simba will have to figure out how to grow up and take back what is rightfully his.",
    "release_date": "2019-07-12"
  }
};

export const mockMovie = {
  "id": 420818,
  "video": false,
  "vote_average": 7.1,
  "title": "The Lion King",
  "popularity": 636.278,
  "poster_path": "\/dzBtMocZuJbjLOXvrl4zGYigDzh.jpg",
  "original_language": "en",
  "original_title": "The Lion King",
  "genre_ids": [
    12,
    16,
    10751,
    18,
    28
  ],
  "backdrop_path": "\/1TUg5pO1VZ4B0Q1amk3OlXvlpXV.jpg",
  "adult": false,
  "overview": "Simba idolises his father, King Mufasa, and takes to heart his own royal destiny. But not everyone in the kingdom celebrates the new cub's arrival. Scar, Mufasa's brother—and former heir to the throne—has plans of his own. The battle for Pride Rock is ravaged with betrayal, tragedy and drama, ultimately resulting in Simba's exile. With help from a curious pair of newfound friends, Simba will have to figure out how to grow up and take back what is rightfully his.",
  "release_date": "2019-07-12"
}

export const mockUser = {
  email: 'happy@sad.com',
  password: 'ilikeburritos',
  id: 2,
  name: 'coolGui420'
}

export const mockGenre = 'popular';

export const mockPost = {
  method: 'POST',
  headers: {
    'Content-Type': 'application/json'
  },
  body: JSON.stringify({
    user: mockUser
  })
}

export const mockLoading = false;

export const mockFavorited = true;

export const mockFavorites = [{
  "id": 420818,
  "video": false,
  "vote_average": 7.1,
  "title": "The Lion King",
  "popularity": 636.278,
  "poster_path": "\/dzBtMocZuJbjLOXvrl4zGYigDzh.jpg",
  "original_language": "en",
  "original_title": "The Lion King",
  "genre_ids": [
    12,
    16,
    10751,
    18,
    28
  ],
  "backdrop_path": "\/1TUg5pO1VZ4B0Q1amk3OlXvlpXV.jpg",
  "adult": false,
  "overview": "Simba idolises his father, King Mufasa, and takes to heart his own royal destiny. But not everyone in the kingdom celebrates the new cub's arrival. Scar, Mufasa's brother—and former heir to the throne—has plans of his own. The battle for Pride Rock is ravaged with betrayal, tragedy and drama, ultimately resulting in Simba's exile. With help from a curious pair of newfound friends, Simba will have to figure out how to grow up and take back what is rightfully his.",
  "release_date": "2019-07-12"
}]

export const mockError = 'Something went wrong';

export const mockGenres = [{
    "id": 28,
    "name": "Action",

  },
  {
    "id": 12,
    "name": "Adventure",

  }, {
    "id": 16,
    "name": "Animation",

  }, {
    "id": 35,
    "name": "Comedy",

  }, {
    "id": 80,
    "name": "Crime",

  }, {
    "id": 99,
    "name": "Documentary",

  }, {
    "id": 18,
    "name": "Drama",

  }, {
    "id": 10751,
    "name": "Family",

  }, {
    "id": 14,
    "name": "Fantasy",

  }, {
    "id": 36,
    "name": "History",

  }, {
    "id": 27,
    "name": "Horror",

  }, {
    "id": 10402,
    "name": "Music",

  }, {
    "id": 9648,
    "name": "Mystery",

  }, {
    "id": 10749,
    "name": "Romance",

  }, {
    "id": 878,
    "name": "Science Fiction",

  }, {
    "id": 10770,
    "name": "TV Movie",

  }, {
    "id": 53,
    "name": "Thriller",

  }, {
    "id": 10752,
    "name": "War",

  }, {
    "id": 37,
    "name": "Western",

  }
]