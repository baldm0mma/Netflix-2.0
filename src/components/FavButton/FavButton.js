import React from 'react';
import { sendFavorite, deleteFavorite, fetchUserFavorites } from '../../utils/API/ApiFetch';
import {setFavorites,chooseMovie } from '../../actions';
// import { chooseMovie, setFavorites, addMovie } from '../../actions';
import { Link, Redirect } from 'react-router-dom';
import filledHeart from '../../images/like-filled.png';
import emptyHeart from '../../images/like-empty.png';
import moreDetails from '../../images/clapperboard.png';
import { connect } from 'react-redux';

const FavButton = ({ movieInfo, user, chooseSpecificMovie, setFavorites }) => {
  const { title, movie_id, isFavorited } = movieInfo;
  const { user_id } = user;

  const toggleFav = async movie => {
    if (user.id) {
      const favorites = await fetchUserFavorites(user.id);
      setFavorites(favorites.data);
      console.log(user.favorites);
      const foundMovie = user.favorites.find(favorite => favorite.movie_id === movie_id);
        if (foundMovie) {
          deleteFavorite(user.id, movie_id);
        } else if (!foundMovie && user.id) {
          const favMovie = {
            ...movieInfo,
            user_id
          }
          sendFavorite(favMovie);
          const favorites = await fetchUserFavorites(user.id);
          setFavorites(favorites.data);
      }
    } else {
      alert('Please log in to favorite a movie')
    }
  };

// export const FavButton = ({ movieData, user_id, currentUserFavorites }) => {
// 	const toggle = currentUserFavorites.find(fav => fav.movie_id === movieData.movie_id);
// 	return (
// 		<button movieData={movieData} user_id={user_id}>
// 			{toggle === -1 ? 'Favorite' : 'Unfavorite'}
// 		</button>
// 	);
// };

// export default FavButton;
  
  return (
    <div className="movie-buttons">
      <Link to={`/movies/${title}`}>
        <button onClick={() => chooseSpecificMovie(title, movie_id)}>
          <img alt="more details" src={moreDetails} />
        </button>
      </Link>
      <button onClick={() => toggleFav(movie_id)}>
        <img
          className="favorite__img-button"
          alt="favorite this movie"
          src={isFavorited ? filledHeart : emptyHeart}
        />
      </button>
    </div>
  );
};

export const mapStateToProps =(state) => ({
  user: state.user,
  specificMovie: state.specificMovie
});

export const mapDispatchToProps=(dispatch) => ({
  chooseSpecificMovie: (title, movie_id) => dispatch(chooseMovie(title, movie_id)),
  setFavorites: favorites => dispatch(setFavorites(favorites))
})

export default connect(mapStateToProps, mapDispatchToProps)(FavButton)
