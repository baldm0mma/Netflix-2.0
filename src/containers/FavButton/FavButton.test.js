import {FavButton, mapDispatchToProps, mapStateToProps} from './FavButton';
import { shallow } from 'enzyme';
import React from 'react';
import { mockUser, mockMovie, mockMovies, mockFavorites } from '../../utils/mockData/mockData';
import {chooseMovie, setFavorites} from '../../actions';

describe('FavButton', () => {
	let wrapper, instance;

	beforeEach(() => {
		wrapper = shallow(
			<FavButton 
				movieInfo={mockMovie} 
				user={mockUser} 
				chooseSpecificMovie={mockMovie}
				setFavorites={mockMovies} />
		);
		instance = wrapper.instance()
	});

	it('should match snapshot', () => {
		expect(wrapper).toMatchSnapshot();
	});

	describe('mapStateToProps', () => {
		it('should return an object with the user and specificMovie', () => {
			const mockState = {
				user: mockUser,
				specificMovie: mockMovie
			};
			const mappedProps = mapStateToProps(mockState);
			expect(mappedProps).toEqual(mockState);
		});
	});

	describe('mapDispatchToProps', () => {
		it('should call dispatch with a chooseMovie action when chooseSpecificMovie is called', () => {
			const mockDispatch = jest.fn();
			const actionToDispatch = chooseMovie(mockMovie.title, mockMovie.id);
			const mappedProps = mapDispatchToProps(mockDispatch);
			mappedProps.chooseSpecificMovie(mockMovie.title, mockMovie.id);
			expect(mockDispatch).toHaveBeenCalledWith(actionToDispatch);
		});
		it('should call dispatch with a setFavorites action when setFavorites is called', () => {
			const mockDispatch = jest.fn();
			const actionToDispatch = setFavorites(mockFavorites);
			const mappedProps = mapDispatchToProps(mockDispatch);
			mappedProps.setFavorites(mockFavorites);
			expect(mockDispatch).toHaveBeenCalledWith(actionToDispatch);
		});
	});
});
