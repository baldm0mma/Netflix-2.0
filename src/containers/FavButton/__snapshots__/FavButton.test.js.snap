// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FavButton should match snapshot 1`] = `undefined`;
